### ISSUE TITLE
CatalogItemsV20220401 type error on getCatalogItem summaries

### ISSUE THREAD

**Issue by jbaelo** [jlevers/selling-partner-api#697]
## Problem description:
Requesting getCatalogItem with CatalogAPI version 20220401 if summaries are included in requested fields we get TypeError on "releaseDate" field.

The response returns a date with 'Y-m-d' format but the deserializer expects format DateTime::RFC3339 ('Y-m-d\TH:i:sP'), this causes DateTime::createFromFormat return false and the releaseDate param in SellingPartnerApi\Seller\CatalogItemsV20220401\Dto\ItemSummaryByMarketplace is hinted as null or DateTime.

All deserializer date or dateTime types uses DateTime::RFC3339 format.

Crescat\SaloonSdkGenerator\Traits\Deserializes line 93

![image](https://github.com/jlevers/selling-partner-api/assets/33716791/ad57c1ec-20f3-4ffc-8f8b-4ab50b1de562)

## Error:
```
SellingPartnerApi\Seller\CatalogItemsV20220401\Dto\ItemSummaryByMarketplace::__construct(): Argument #15 ($releaseDate) must be of type ?DateTime, bool given, 
called in vendor/highsidelabs/saloon-sdk-generator/src/Traits/Deserializes.php on line 74
```

## Code

```php
$connector = SellingPartnerApi::make(/* ... */)->seller();
$response = $connector->catalogItemsV20220401()->getCatalogItem('ASIN', ['MARKETPLACE_ID']);
```

**Comment by ssamran15** [comment]
@jbaelo, were you able to resolve this?

**Comment by jbaelo** [comment]
> @jbaelo, were you able to resolve this?

If you use CatalogItemsV20201201 to request sumaries this wont fail, anyway still waiting for the fix

**Comment by jbaelo** [comment]
@ssamran15 use CatalogItemsV20201201 to ask for summaries, that isn't broken.

This bug refers to highsidelabs/saloon-sdk-generator so still waiting for the author to fix it.

**Comment by jlevers** [comment]
This should be fixed in v7, which will be released shortly.

**Comment by jlevers** [comment]
Released in `v7.0.0`.
